src/mcts/search.cc

```diff
@@ -1291,9 +1302,20 @@ void SearchWorker::ExtendNode(Node* node) {
       return;
     }
 
-    if (history_.Last().GetRepetitions() >= 2) {
+    const auto repetitions = history_.Last().GetRepetitions();
+    // Mark two-fold repetitions as draws according to settings.
+    // Depth starts with 1 at root, so number of plies in PV is depth - 1.
+    if (repetitions >= 2) {
       node->MakeTerminal(GameResult::DRAW);
       return;
+    } else if (repetitions == 1 && params_.GetTwoFoldDraws() &&
+               depth - 1 >= 4 &&
+               depth - 1 >= history_.Last().GetPliesSincePrevRepetition()) {
+      const auto cyclelength = history_.Last().GetPliesSincePrevRepetition();
```

Comment:
At least clang seems to directly reuse this value now because it's just a cached value on the position instead of computed.

Reply:
Thanks for checking, I appreciate it!